Improve pull request template

yubiuser opened the pull request:
**What does this PR aim to accomplish?:**

1) Add info about commit signing (required by `core` and `FTL`)
2) Add requirements taken from the FTL PULL_REQUEST template

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

rdwebdesign commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -7,6 +7,7 @@ We are all volunteers and completing the process outlined will help us review yo
+ 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 
 ---
 - **What does this PR aim to accomplish?:**
```
I think this dash is not intended:

```suggestion
**What does this PR aim to accomplish?:**
```

yubiuser replied:
Removed it for all three items

PromoFaux commented:
I think maybe we could use HTML comments instead of the `replace this text with`, like with the Docker template:

https://raw.githubusercontent.com/pi-hole/docker-pi-hole/master/.github/PULL_REQUEST_TEMPLATE.md

They don't show up in the final issue.

They also don't show up in the mattermost github plugin:
![image](https://user-images.githubusercontent.com/1998970/200118364-07b6395e-2334-4737-8864-f58382678961.png)


They do show up for Gerald, though - but that's down to lazy bot-programming on my part

![image](https://user-images.githubusercontent.com/1998970/200118383-9a60f860-1141-460c-b0fd-2500a7e6525c.png)

yubiuser commented:
Nice idea. I added HTML comments.

Additionally, I added a point that could cause some discussion: I changed the paragraph about "what documentation changes are necessary" to a "link all documentation PRs". I think if someone is submitting a PR that requires a change in documentation this should be supplied as well. We should not postpone documentation to an infinite time "later". At some point, someone needs to do the housekeeping. I believe, it's done best at the moment the change is made.

PromoFaux commented:
Agree with the point about doc PRs 👍
